mitmproxy/protocol/http.py

```diff
@@ -359,7 +364,10 @@ def establish_server_connection(self, flow):
                 self.set_server(address, tls, address.host)
             # Establish connection is neccessary.
             if not self.server_conn:
-                self.connect()
```

Comment:
We need to fail if there is an error with the server connection. Just returning False is not an option I think.